frontend/appflowy_flutter/lib/ai/service/select_model_bloc.dart

```diff
@@ -1,82 +1,113 @@
 import 'dart:async';
 
-import 'package:appflowy/ai/service/ai_input_control.dart';
+import 'package:appflowy/ai/service/ai_model_state_notifier.dart';
 import 'package:appflowy_backend/dispatch/dispatch.dart';
 import 'package:appflowy_backend/protobuf/flowy-ai/entities.pbserver.dart';
+import 'package:equatable/equatable.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
-import 'package:protobuf/protobuf.dart';
 
 part 'select_model_bloc.freezed.dart';
 
+class AiModel extends Equatable {
+  const AiModel({
+    required this.name,
+    required this.isLocal,
+  });
+
+  factory AiModel.fromPB(AIModelPB pb) {
+    return AiModel(name: pb.name, isLocal: pb.isLocal);
+  }
+
+  AIModelPB toPB() {
+    return AIModelPB()
+      ..name = name
+      ..isLocal = isLocal;
+  }
+
+  final String name;
+  final bool isLocal;
+
+  @override
+  List<Object?> get props => [name, isLocal];
+}
+
 class SelectModelBloc extends Bloc<SelectModelEvent, SelectModelState> {
   SelectModelBloc({
-    required this.objectId,
-  })  : _aiModelStateNotifier = AIModelStateNotifier(objectId: objectId),
-        super(const SelectModelState()) {
-    _aiModelStateNotifier.init();
-    _aiModelStateNotifier.startListening(
-      onAvailableModelsChanged: (models) {
-        if (!isClosed) {
-          add(SelectModelEvent.didLoadModels(models));
-        }
-      },
-    );
-
+    required AIModelStateNotifier aiModelStateNotifier,
+  })  : _aiModelStateNotifier = aiModelStateNotifier,
```

Comment:
TBH, the notification stuff is very light weight. We don't need to worry about its performance.

Reply:
https://github.com/user-attachments/assets/2a5caee2-ea55-4617-8560-04bf0d6fc367

Reply:
Ok. It likes a UI refresh issues. Because it's async action. It doesn't releate to performance. And yes, by using sharing instance can fix this UI flashing